webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.settings.ts

```diff
@@ -118,3 +118,81 @@ export const configForExperiment = (id: number): SettingsConfig<Settings> => ({
   },
   storagePath: `${id}-checkpoints`,
 });
+
+export const configForTrial = (id: number): SettingsConfig<Settings> => ({
+  settings: {
+    columns: {
+      defaultValue: DEFAULT_COLUMNS,
+      skipUrlEncoding: true,
+      storageKey: 'columns',
+      type: array(
+        union([
+          literal('action'),
+          literal('uuid'),
+          literal('state'),
+          literal('searcherMetric'),
+          literal('totalBatches'),
+          literal('checkpoint'),
+        ]),
+      ),
+    },
+    columnWidths: {
+      defaultValue: DEFAULT_COLUMNS.map((col: CheckpointColumnName) => DEFAULT_COLUMN_WIDTHS[col]),
+      skipUrlEncoding: true,
+      storageKey: 'columnWidths',
+      type: array(number),
+    },
+    row: {
+      defaultValue: undefined,
+      skipUrlEncoding: true,
+      storageKey: 'row',
+      type: union([undefinedType, array(string)]),
+    },
+    sortDesc: {
+      defaultValue: true,
+      storageKey: 'sortDesc',
+      type: boolean,
+    },
+    sortKey: {
+      defaultValue: Checkpointv1SortBy.UUID,
+      storageKey: 'sortKey',
+      type: union([
+        literal(Checkpointv1SortBy.BATCHNUMBER),
+        literal(Checkpointv1SortBy.ENDTIME),
+        literal(Checkpointv1SortBy.SEARCHERMETRIC),
+        literal(Checkpointv1SortBy.STATE),
+        literal(Checkpointv1SortBy.TRIALID),
+        literal(Checkpointv1SortBy.UNSPECIFIED),
+        literal(Checkpointv1SortBy.UUID),
+      ]),
+    },
+    state: {
+      defaultValue: undefined,
+      storageKey: 'state',
+      type: union([
+        undefinedType,
+        array(
+          union([
+            literal(CheckpointState.Active),
+            literal(CheckpointState.Completed),
+            literal(CheckpointState.Deleted),
+            literal(CheckpointState.PartiallyDeleted),
+            literal(CheckpointState.Error),
+            literal(CheckpointState.Unspecified),
+          ]),
```

Comment:
nit: same as the above re: valueof
```suggestion
          valueof(CheckpointState),
```